utilities/titandb/titan_db_test.cc

```diff
@@ -361,6 +363,167 @@ TEST_F(TitanDBTest, DISABLED_ReadAfterDropCF) {
   }
 }
 
+TEST_F(TitanDBTest, BlobFileIOError) {
+  std::unique_ptr<FaultInjectionTestEnv> mock_env(
+      new FaultInjectionTestEnv(env_));
+  options_.env = mock_env.get();
+  Open();
+
+  std::map<std::string, std::string> data;
+  const int kNumEntries = 100;
+  for (uint64_t i = 1; i <= kNumEntries; i++) {
+    Put(i, &data);
+  }
+  ASSERT_EQ(kNumEntries, data.size());
+  CompactRangeOptions copts;
+  ASSERT_OK(db_->CompactRange(copts, nullptr, nullptr));
+  VerifyDB(data);
+
+  // not sync at reader
+  SyncPoint::GetInstance()->SetCallBack(
+    "BlobStorage::FindFile", [&](void *) {
+      mock_env->SetFilesystemActive(false, Status::IOError("Injected error"));
+    });
+  SyncPoint::GetInstance()->EnableProcessing();
+  for(auto& it : data) {
+    std::string value;
+    if(it.second.size() > options_.min_blob_size) {
+      ASSERT_FALSE(db_->Get(ReadOptions(), it.first, &value).ok());
+      mock_env->SetFilesystemActive(true);
+    }
+  }
+  std::unique_ptr<Iterator> iter(db_->NewIterator(ReadOptions()));
+  iter->SeekToFirst();
+  mock_env->SetFilesystemActive(true);
+  for (const auto& it : data) {
+    if(iter->Valid()) {
```

Comment:
since `SeekToFirst` will see IO error, the iterator will never be valid. We need to test both scenarios: (1) IO error on `SeekXXX`, (2) no error on `SeekXXX` but IO error on `Next`.